src/test/java/seedu/application/logic/commands/ListCommandTest.java

```diff
@@ -28,12 +33,64 @@ public void setUp() {
 
     @Test
     public void execute_listIsNotFiltered_showsSameList() {
-        assertCommandSuccess(new ListCommand(), model, ListCommand.MESSAGE_SUCCESS, expectedModel);
+        assertCommandSuccess(new ListCommand(new FieldComparator(FieldComparator.EMPTY_COMPARATOR_SPECIFIER)),
+            model, ListCommand.MESSAGE_SUCCESS, expectedModel);
     }
 
     @Test
     public void execute_listIsFiltered_showsEverything() {
         showJobAtIndex(model, INDEX_FIRST_JOB);
-        assertCommandSuccess(new ListCommand(), model, ListCommand.MESSAGE_SUCCESS, expectedModel);
+        assertCommandSuccess(new ListCommand(new FieldComparator(FieldComparator.EMPTY_COMPARATOR_SPECIFIER)),
+            model, ListCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_listIsSortedByRoles_showsSortedListByRoles() {
+        FieldComparator fieldComparator = new FieldComparator(ROLE_SPECIFIER);
+        expectedModel.sortJobs(fieldComparator);
+        assertCommandSuccess(new ListCommand(fieldComparator), model, ListCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_listIsSortedByCompany_showsSortedListByCompany() {
+        FieldComparator fieldComparator = new FieldComparator(COMPANY_SPECIFIER);
+        expectedModel.sortJobs(fieldComparator);
+        assertCommandSuccess(new ListCommand(fieldComparator), model, ListCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_listIsSortedByDeadline_showsSortedListByDeadline() {
+        FieldComparator fieldComparator = new FieldComparator(Deadline.DEADLINE_SPECIFIER);
+        expectedModel.sortJobs(fieldComparator);
+        assertCommandSuccess(new ListCommand(fieldComparator), model, ListCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_listIsSortedByStatus_showsSortedListStatus() {
```

Comment:
Can change the name to showsSortedListByStatus to match with the others :)

Reply:
Good spotting! I have changed that!